tests/sklearn/test_dump_onnx.py

```diff
@@ -457,4 +405,106 @@ def test_dump(
+        model.compile(x, default_configuration)
+
+    # Get ONNX model
+    onnx_model = model.onnx_model
+
+    # Remove initializers, since they change from one seed to the other
+    model_name = get_model_name(model_class)
+    if model_name in [
+        "DecisionTreeRegressor",
+        "DecisionTreeClassifier",
+        "RandomForestClassifier",
+        "RandomForestRegressor",
+        "XGBClassifier",
+        "KNeighborsClassifier",
+    ]:
+        while len(onnx_model.graph.initializer) > 0:
+            del onnx_model.graph.initializer[0]
+
+    str_model = onnx.helper.printable_graph(onnx_model.graph)
+    print(f"\nCurrent {model_name=}:\n{str_model}")
+    print(f"\nExpected {model_name=}:\n{str_expected}")
+
+    # Test equality when it does not depend on seeds
+    # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/3266
+    if not is_model_class_in_a_list(model_class, _get_sklearn_tree_models(select="RandomForest")):
+        # The expected graph is usually a string and we therefore directly test if it is equal to
+        # the retrieved graph's string. However, in some cases such as for TweedieRegressor models,
+        # this graph can slightly changed depending on some input's values. We then expected the
+        # string to match as least one of them expected strings (as a list)
+        if isinstance(str_expected, str):
+            assert str_model == str_expected
+        else:
+            assert str_model in str_expected
+
+
+@pytest.mark.parametrize("model_class, parameters", UNIQUE_MODELS_AND_DATASETS)
+def test_dump(
+    model_class,
+    parameters,
+    load_data,
+    default_configuration,
+):
+    """Tests dump."""
+
+    model_name = get_model_name(model_class)
+
+    # Some models have been done with different n_classes which create different ONNX
+    if parameters.get("n_classes", 2) != 2 and model_name in ["LinearSVC", "LogisticRegression"]:
+        return
+
+    if model_name == "NeuralNetClassifier":
+        model_class = partial(
+            NeuralNetClassifier,
+            module__n_layers=3,
+            module__power_of_two_scaling=False,
+            max_epochs=1,
+            verbose=0,
+            callbacks="disable",
+        )
+    elif model_name == "NeuralNetRegressor":
+        model_class = partial(
+            NeuralNetRegressor,
+            module__n_layers=3,
+            module__n_w_bits=2,
+            module__n_a_bits=2,
+            module__n_accum_bits=7,  # Stay with 7 bits for test exec time
+            module__n_hidden_neurons_multiplier=1,
+            module__power_of_two_scaling=False,
+            max_epochs=1,
+            verbose=0,
+            callbacks="disable",
+        )
+
+    check_onnx_file_dump(model_class, parameters, load_data, default_configuration)
+
+    # Additional tests exclusively dedicated for tree ensemble models.
+    if model_class in _get_sklearn_tree_models()[2:]:
```

Comment:
I'm not sure what is this `[2:]` ? Is it in order to avoid decision tree models ? In that cas could you do `_get_sklearn_tree_models(select=["RandomForest", "XGB"])` instead ? it's more explicit that way

Reply:
thanks, it was a mistake !

Reply:
thanks, it was a mistake !